### ISSUE TITLE
Fix Mujoco dlopen issue

### ISSUE THREAD

**Issue by sidor926** [Zhourobotics/outdoor_slam_ros2#34]
Fix Mujoco dlopen issue~~Add mujoco2.3.2 support to the repository.~~ 
/CC: @adeeb10abbas

**Comment by adeeb10abbas** [comment]
Any updates on this?

**Comment by sidor926** [comment]
Working on this!

**Comment by adeeb10abbas** [comment]
Fyi - 
Here's the detailed issue - 
trying to use the [latest mujoco (Python3)](https://pypi.org/project/mujoco/) in a Bazel workspace via rules_python but I run into an Invalid ELF Header error
This is what my WORKSPACE looks like -
```
# All the Python Stuff ##
load("@rules_python//python:repositories.bzl", "python_register_toolchains")
python_register_toolchains(
    name = "python3_10",
    python_version = "3.10",
)

load("@python3_10//:defs.bzl", "interpreter")
load("@rules_python//python:pip.bzl", "pip_install")

pip_install(
    name = "pip",
    requirements = "//:requirements.txt",
)
load("@pip//:requirements.bzl", "install_deps")
# Initialize repositories for all packages in requirements.txt.
install_deps()
## End Python Stuff ##
```
And the requirements.txt has the Mujoco version 2.3.2. Using the setup shown it works for me up until 2.2.x but starting with 2.3.x it throws the invalid ELF header error. The pip install outside of a bazel workspace works fine on all platforms, did this to make sure the upstream package wasn't broken to begin with. Any help would be appreciated, I can also share a small repro of the issue if needed.
Edit: Platforms tested on - Jammy (x86_64) and M1 Mac

More about the output - 

```
abbas@abbas-XPS-15-9520:~/src/dragon/upstream/outdoor_slam_ros2$ bazel run //example/sample_py:test_py -j 12
INFO: Analyzed target //example/sample_py:test_py (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //example/sample_py:test_py up-to-date:
  bazel-bin/example/sample_py/_test_py_shim.py
  bazel-bin/example/sample_py/test_py
INFO: Elapsed time: 339.878s, Critical Path: 128.51s
INFO: 1128 processes: 301 internal, 827 linux-sandbox.
INFO: Build completed successfully, 1128 total actions
INFO: Running command line: bazel-bin/example/sample_py/test_py
Traceback (most recent call last):
  File "/home/abbas/.cache/bazel/_bazel_abbas/438cf2d339a3e2212999f1c46d280f19/execroot/outdoorSLAM/bazel-out/k8-opt/bin/example/sample_py/test_py.runfiles/outdoorSLAM/example/sample_py/test_py.py", line 7, in <module>
    import mujoco
  File "/home/abbas/.cache/bazel/_bazel_abbas/438cf2d339a3e2212999f1c46d280f19/execroot/outdoorSLAM/bazel-out/k8-opt/bin/example/sample_py/test_py.runfiles/pip_mujoco/site-packages/mujoco/__init__.py", line 59, in <module>
    _load_all_bundled_plugins()
  File "/home/abbas/.cache/bazel/_bazel_abbas/438cf2d339a3e2212999f1c46d280f19/execroot/outdoorSLAM/bazel-out/k8-opt/bin/example/sample_py/test_py.runfiles/pip_mujoco/site-packages/mujoco/__init__.py", line 57, in _load_all_bundled_plugins
    PLUGIN_HANDLES.append(ctypes.CDLL(os.path.join(directory, filename)))
  File "/usr/lib/python3.10/ctypes/__init__.py", line 374, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: /home/abbas/.cache/bazel/_bazel_abbas/438cf2d339a3e2212999f1c46d280f19/execroot/outdoorSLAM/bazel-out/k8-opt/bin/example/sample_py/test_py.runfiles/pip_mujoco/site-packages/mujoco/plugin/__init__.py: invalid ELF header
```

After talking to Jeremy from TRI (He's a god) - He says: 
```
Second best guess -- it's trying to dlopen the wrong file, something like grabbing a python source file and trying to load it as a native module. I would suggest manually downloading the mujoco whl files for the working and not-working releases, and see if anything inside of them got shuffled around. (The whl are just zip files, you can peek inside easily, without pip.) 
To be clear -- I am sorry it's not working for you but one of my personal BIGGEST PET PEEEVES is libraries that dlopen plugins.  Something always goes wrong.  I just hate that as a software architecture. (edited) 
```

And then... Mujoco released `2.3.3` which solved this misplaced weird dlopen issue! Problem solved. It was not my Bazel workspace at all!

**Comment by adeeb10abbas** [comment]
Closing it now, since the following works perfectly - 

```
abbas@abbas-XPS-15-9520:~/src/dragon/upstream/outdoor_slam_ros2$ bazel run //example/sample_py:test_py -j 12
INFO: Analyzed target //example/sample_py:test_py (2 packages loaded, 48 targets configured).
INFO: Found 1 target...
Target //example/sample_py:test_py up-to-date:
  bazel-bin/example/sample_py/_test_py_shim.py
  bazel-bin/example/sample_py/test_py
INFO: Elapsed time: 3.057s, Critical Path: 0.84s
INFO: 5 processes: 5 internal.
INFO: Build completed successfully, 5 total actions
INFO: Running command line: bazel-bin/example/sample_py/test_py
Websockets version: 10.4
the opencv version is 4.7.0
the mujoco version is 2.3.3
the numpy version is 1.24.0
Random operation with pydrake: 500000.0
```
